### PR TITLE
Reorder columns

### PR DESCRIPTION
**Pull request by deaspo**

### CODE REVIEW

**Comment by deaspo** [comment]
@dannyparsons though this dialog works am not contented  Could we have another user control specifically for this purpose, where we can reorder by moving columns up and down, Something like below:
suggestion

**Comment by dannyparsons** [comment]
Yes something to move columns up and down is definitely what we want. Attached is the dialog me and Roger came up with. There is just a data frame selector and then all the columns are shown in their current order.
We then thought generally about where else we might need something like this: reorder factor levels, reorder data frames, manage models, manage graphs etc.
This led us to thinking that all these 'lists' could be displayed better within a grid as sometimes we may want multiple columns in the list, sometimes we want to edit it, sometimes read only etc.

So we suggest a new user control that is just a grid that fills the control. The control can then have methods like fill grid, set read only/editable, hide/show the tabs etc.
Then there's a second control that inherits from this control and also has the four arrow buttons (up, down and the double arrows move to top and bottom). So the arrows move the rows of the grid up and down. That's where we will need some code for swapping columns in a grid.

Quite a lot of dialogs will then use these controls for displaying lists in a grid. Do you want to create these two controls?

reorder

**Comment by deaspo** [comment]
Yes I would like to, but I have a questions. Why put it in form of a grid? I would prefer list view.

**Comment by dannyparsons** [comment]
Good question. I discussed with Roger and a list view seems sensible for the simple reordering of columns, data frames etc. So then it's just one user control with a list view and arrow buttons.
Some factor dialogs may still use a grid but we can decide on those details later.

**Comment by deaspo** [comment]
Question, how do I TestOkEnabled with the new user control Reorder?

**Comment by dannyparsons** [comment]
Good question. What we need is a function of the reorder control that
returns True or False and use this for the TestOK. There's a few different
ways that function could work. It could simply test if the list view is
empty or not. Or a bit more complicated, it could test if the items in the
list are in a different order to when it started, and we say that Ok isn't
enabled if the list is in the same order as the start.
Either would be fine for now so happy to start with the simple
implementation so that you don't have to spend a lot of time getting it
working but up to you.

**Comment by deaspo** [comment]
Updated it with new user control reorder

File: instat/ucrReorder.vb [hunk 1]
```diff
@@ -145,7 +145,7 @@ Public Class ucrReorder
     Public Sub loadList()
         Dim dfTemp As CharacterMatrix
         Select Case strDataType
-            Case "column"
+            Case "all"
                 If ucrDataFrameList IsNot Nothing AndAlso ucrDataFrameList.cboAvailableDataFrames.Text <> "" Then
                     frmMain.clsRLink.clsEngine.Evaluate(ucrDataFrameList.cboAvailableDataFrames.SelectedItem & "=" & frmMain.clsRLink.strInstatDataObject & "$get_variables_metadata(data_name = " & Chr(34) & ucrDataFrameList.cboAvailableDataFrames.SelectedItem & Chr(34) & ", data_type = " & Chr(34) & strDataType & Chr(34) & ")[,1]")
```
Comment (dannyparsons) [inline instat/ucrReorder.vb]: Alternatively, you could use the get_column_names method in the Instat object. This just takes a single argument for the data_name.

**Comment by deaspo** [comment]
Made the modifications.

**Pull request merged**